addons/mail/static/src/core/web/thread_service_patch.js

```diff
@@ -191,10 +191,12 @@ patch(ThreadService.prototype, {
     open(thread, replaceNewMessageChatWindow, options) {
         if (!this.store.discuss.isActive && !this.ui.isSmall) {
             this._openChatWindow(thread, replaceNewMessageChatWindow, options);
+            this.store.env.services["bus_service"].addChannel(thread.busChannel);
             return;
         }
         if (this.ui.isSmall && thread.model === "discuss.channel") {
             this._openChatWindow(thread, replaceNewMessageChatWindow, options);
+            this.store.env.services["bus_service"].addChannel(thread.busChannel);
             return;
```

Comment:
As this LOC is common for both cases and `_openChatWindow` is a private method for this patch only which will be used here only. You can move this LOC into that method.